### PR TITLE
em-820 update hrefs to use new documentUrl, contentUrl, and public project url

### PR DESCRIPTION
**Pull request by btzc**
Change hrefs for the public project details page, the home page, and the news page to use the window's host as a host instead of hard coding it.

Did some url parsing to add window's host to front of url as long as documentUrl or projectUrl is a route and not a full url (doesn't include http://www).

### CODE REVIEW

**Comment by NickPhura** [comment]
We need to update the unit tests for any changes here as well.

File: src/app/home/home.component.ts [hunk 1]
```diff
@@ -15,9 +15,24 @@ export class HomeComponent implements OnInit {
...
     );
-    this.hostname = this.api.hostnameEPIC;
+  }
+
+  setDocumentUrl(data) {
+    data.forEach(activity => {
+      if(!activity.documentUrl) {
+        return ;
+      }
+      if (activity.documentUrl.includes('http://www') || activity.documentUrl.length === 0) {
```
Comment (esune) [inline src/app/home/home.component.ts]: I am not too sure about what we're trying to do here:
* ```activity.documentUrl.length === 0``` should be already caught by the check at line 28
* we are setting ```activity.documentUrl``` to be...itself?
* it looks like at the very least we could change this to something like
```
if (!activity.documentUrl.includes('http://www'){
  activity.documentUrl = `${this.api.hostnameEPIC }${ activity.documentUrl }`;
}
```

Maybe I am missing/forgetting something here though...

File: src/app/news/news.component.ts [hunk 2]
```diff
@@ -43,7 +44,20 @@ export class NewsComponent implements OnInit {
...
-    this.hostname = this.api.hostnameEPIC;
+  }
+
+  setDocumentUrl(data) {
+    // attach host to document url if it goes to esm-server
+    data.forEach(activity => {
+      if(!activity.documentUrl) {
+        return ;
+      }
+      if (activity.documentUrl.includes('http://www') || activity.documentUrl.length === 0) {
```
Comment (esune) [inline src/app/news/news.component.ts]: Same as my comment in _home.component.ts_

File: src/app/project-detail/project-detail.component.ts [hunk 3]
```diff
@@ -50,6 +53,18 @@ export class ProjectDetailComponent implements OnInit {
       error => console.log(error)
     );
   }
+  setDocumentUrl(project) {
+    project.recent_activities.forEach(activity => {
+      if(!activity.documentUrl) {
+        return ;
+      }
+      if (activity.documentUrl.includes('http://www') || activity.documentUrl.length === 0) {
```
Comment (esune) [inline src/app/project-detail/project-detail.component.ts]: Here too, same as above comment(s).

File: src/app/home/home.component.ts (hunk 1 above)
Comment (btzc) [reply src/app/home/home.component.ts]: Fixed all

File: src/app/project-detail/project-detail.component.ts [hunk 4]
```diff
@@ -50,6 +53,17 @@ export class ProjectDetailComponent implements OnInit {
       error => console.log(error)
     );
   }
+  setDocumentUrl(project) {
+    const regex = /http(s)?:\/\/(www.)?/;
+    project.recent_activities.forEach(activity => {
+      if (!activity.documentUrl) {
+        return ;
+      }
+      if (!regex.test(activity.documentUrl)) {
```
Comment (esune) [inline src/app/project-detail/project-detail.component.ts]: Do we really need to check the document URLs and set them to point to `hostnameEPIC` if we update the data on the database ([see here](https://github.com/bcgov/esm-server/pull/387))?

**Pull request merged**